no-downtime-upgrade/src/main/java/org/keycloak/playground/nodowntimeupgrade/naive_jackson/Migration.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.playground.nodowntimeupgrade.naive_jackson;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ *
+ * @author hmlnarik
+ */
+public class Migration {
+
+    private static final List<Function<ObjectNode, ObjectNode>> MIGRATORS = Arrays.asList(
+      o -> o,
+      Migration::migrateTreeFrom1To2,
+      Migration::migrateTreeFrom2To3,
+      Migration::migrateTreeFrom3To4
+    );
+
+    public static ObjectNode migrateTreeFrom1To2(ObjectNode node) {
```

Comment:
Nitpick: It seems that method "migrateTreeFromXToY" can be private?

Reply:
Sure :-) the `naive_jackson` package is intentionally left unoptimal and literally naive to provide the sample of how to implement a store that passes all the tests, but not to provide premature solutions and skew ideas that can stem from this exercise.